std/range/package.d

```diff
@@ -5277,24 +5278,23 @@ if ((isIntegral!(CommonType!(B, E)) || isPointer!(CommonType!(B, E)))
 
         this(Value current, Value pastLast, StepType step)
         {
-            if ((current < pastLast && step >= 0) ||
-                    (current > pastLast && step <= 0))
+            if ((current < pastLast && step > 0) ||
+                    (current > pastLast && step < 0))
             {
                 this.step = step;
                 this.current = current;
                 if (step > 0)
                 {
                     assert(unsigned((pastLast - current) / step) <= size_t.max);
-
-                    this.pastLast = pastLast - 1;
+                    // Cast below can't fail because current < pastLast
+                    this.pastLast = cast(Value) (pastLast - 1);
```

Comment:
It's a a toss seeing that this variant has more duplication. I'll try a variant with early returns.